app/models/miq_expression.rb

```diff
@@ -673,18 +673,22 @@ def _to_sql(exp, tz)
 
       operands = self.class.operands2sqlvalue(operator, exp[operator])
       clause = operands.join(" #{self.class.normalize_sql_operator(operator)} ")
-    when "like", "not like", "starts with", "ends with", "includes"
-      operands = self.class.operands2sqlvalue(operator, exp[operator])
-      case operator.downcase
-      when "starts with"
-        operands[1] = "'" + operands[1].to_s + "%'"
-      when "ends with"
-        operands[1] = "'%" + operands[1].to_s + "'"
-      when "like", "not like", "includes"
-        operands[1] = "'%" + operands[1].to_s + "%'"
-      end
-      clause = operands.join(" #{self.class.normalize_sql_operator(operator)} ")
-      clause = "!(" + clause + ")" if operator.downcase == "not like"
+    when "like", "includes"
+      field = Field.parse(exp[operator]["field"])
+      table = Arel::Table.new(field.table_name)
+      clause = table[field.column].matches(Arel::Nodes::Quoted.new("%#{exp[operator]["value"]}%")).to_sql
```

Comment:
1. Any change of removing `to_sql` in the near future?
2. Would you define `Field#arel_attribute`? [usage example]
3. Could you move the matching / quoting into a helper method?

```ruby
class Field
  # def arel_table ; model.arel_table ; end
  def arel_attribute
    model.arel_attribute(field.column)
  end

  def matches(like_exp)
    like_expr = Arel::Nodes::Quoted.new(like_expr) if like_expr.kind_of?(String)
    arel_attribute.matches(like_expr)
end

field = Field.parse(exp[operator]["field"])
clause = field.matches("%#{exp[operator]["value"]}%").to_sql
```



fun thought: also, arel has some pretty sweet math functions like `arel_table[:column] / 2 + 5`

[usage example]: https://github.com/ManageIQ/manageiq/pull/7813/files#diff-74635bd5d74cef5b820c9c144dc62592R56